src/components/ScoreSetHeatmap.vue

```diff
@@ -3,22 +3,27 @@
 
     <div style="text-align: center;">Functional Score by Variant</div>
     <div class="mave-heatmap-wrapper">
-      <div id="mave-heatmap-container" class="heatmapContainer" ref="heatmapContainer">
-        <div id="mave-heatmap-scroll-container" class="heatmapScrollContainer" ref="heatmapScrollContainer">
-          <div id="mave-stacked-heatmap-container" class="mave-simple-variants-stacked-heatmap-container" ref="simpleVariantsStackedHeatmapContainer" />
-          <div id="mave-variants-heatmap-container" class="mave-simple-variants-heatmap-container" ref="simpleVariantsHeatmapContainer" />
+      <template v-if="showHeatmap">
+        <div id="mave-heatmap-container" class="heatmapContainer" ref="heatmapContainer">
+          <div id="mave-heatmap-scroll-container" class="heatmapScrollContainer" ref="heatmapScrollContainer">
+            <div id="mave-stacked-heatmap-container" class="mave-simple-variants-stacked-heatmap-container" ref="simpleVariantsStackedHeatmapContainer" />
+            <div id="mave-variants-heatmap-container" class="mave-simple-variants-heatmap-container" ref="simpleVariantsHeatmapContainer" />
+          </div>
         </div>
-      </div>
-      <div class="mave-heatmap-controls">
-        <span class="mave-heatmap-controls-title">Heatmap format</span>
-        <SelectButton
-          v-model="layout"
-          :allow-empty="false"
-          option-label="title"
-          option-value="value"
-          :options="[{title: 'Normal', value: 'normal'}, {title: 'Compact', value: 'compact'}]"
-        />
-      </div>
+        <div class="mave-heatmap-controls">
+          <span class="mave-heatmap-controls-title">Heatmap format</span>
+          <SelectButton
+            v-model="layout"
+            :allow-empty="false"
+            option-label="title"
+            option-value="value"
+            :options="[{title: 'Normal', value: 'normal'}, {title: 'Compact', value: 'compact'}]"
+          />
+        </div>
+      </template>
+      <template v-else>
+        <div class="no-heatmap-message">Insufficient score data to generate a heatmap.</div>
+      </template>
```

Comment:
I think the only other thing is we should not show this message if the score set is published. So we would have a template that looks like the following. I also added another line to the heatmap message so that users also know the reason we cannot generate a heatmap for them.
```
<template v-if="showHeatmap">
    ...
</template>
<template v-else-if="scoreSet.private">
    <div class="no-heatmap-message">Insufficient score data to generate a heatmap. A variant should be present at at least 5% of possible positions to generate a heatmap.</div>
</template>
```